src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -28,6 +46,7 @@ public class TagHelperDescriptor
         /// Instantiates a new instance of the <see cref="TagHelperDescriptor"/> class with the given
         /// <paramref name="attributes"/>.
         /// </summary>
+        /// <param name="prefix">The prefix of the HTML element name the tag helper targets.</param>
```

Comment:
thinking about this a bit and talking a couple of things through w/ @rynowak, I realize you've got two uses of these descriptors.  on one hand you have descriptor resolution, where the match between add/removeTagHelper directives and discovered tag helpers should use the original (from the tag helper) tag name.  on the other hand, the rewriter needs to match Razor tags against the full name of a tag helper.  note the prefix serves neither purpose and causes repeated string concatenations.

I suggest:
- add the `Prefix` to the `TagHelperDescriptorResolutionContext` (or another high-level context) so tooling can use it (if necessary)
- add a `string ShortName { get; }` property to `TagHelperDescriptor` and change the tag helper filtering code to use that instead.  seems like the less-used `TagName` requirement.  but flip it around to be `TagName` and `FullName` if that minimizes the churn.